tests/e2e/support/page/publicLinkPage.ts

```diff
@@ -0,0 +1,55 @@
+import { BrowserContext, Page } from 'playwright'
+import { state } from '../../cucumber/environment/shared'
+import { lastCreatedPublicLink } from './files/publicLink'
+import { expect } from '@playwright/test'
+import util = require('util')
+
+export class PublicLinkPage {
+  public context: BrowserContext
+  private static page: Page
+  private readonly passwordInput: string
+  private readonly passwordSubmitButton: string
+  private readonly fileUploadInput: string
+  private uploadedFiles: string
+
+  constructor() {
+    this.passwordInput = 'input[type="password"]'
+    this.passwordSubmitButton =
+      '//*[@id="password-submit"]|//*[@id="oc-textinput-3"]/ancestor::div[contains(@class, "oc-mb-s")]/following-sibling::button'
+    this.uploadedFiles = `text='%s'`
+    this.fileUploadInput = '//input[@id="file_upload_start" or @class="dz-hidden-input"]'
+  }
+
+  async setup(): Promise<void> {
+    this.context = await state.browser.newContext({ ignoreHTTPSErrors: true })
+    PublicLinkPage.page = await this.context.newPage()
+  }
+
+  async navigateToPublicLink(): Promise<void> {
+    await this.setup()
+    await PublicLinkPage.page.goto(lastCreatedPublicLink)
+  }
+
+  async authenticatePassword(password: any): Promise<void> {
+    await PublicLinkPage.page.locator(this.passwordInput).fill(password)
+    await PublicLinkPage.page.locator(this.passwordSubmitButton).click()
+  }
+
+  async isFileVisible(actionType: string, file: string): Promise<void> {
+    if (actionType === 'be') {
+      await expect(PublicLinkPage.page.locator(util.format(this.uploadedFiles, file))).toBeVisible()
+    } else {
+      await expect(
+        PublicLinkPage.page.locator(util.format(this.uploadedFiles, file))
+      ).not.toBeVisible()
+    }
+  }
+
+  async uploadFiles(files: any): Promise<void> {
+    await PublicLinkPage.page.locator(this.fileUploadInput).setInputFiles(files.path)
+  }
```

Comment:
avoid using `any`
you only need filePath here so can be refactored asbelow and pass path to the function call:
```suggestion
  async uploadFiles(filePath: string): Promise<void> {
    await PublicLinkPage.page.locator(this.fileUploadInput).setInputFiles(filePath)
  }
```